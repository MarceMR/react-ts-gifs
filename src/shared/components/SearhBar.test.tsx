import { describe, expect, test, vi } from "vitest";
import { SearchBar } from "./SearchBar";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";

describe('Pruebas en SearhBar', () => {

    test('Should render component properly', () => {
        const { container } = render(<SearchBar onQuery={() => { }} />)
        expect(container).toMatchSnapshot();
        expect(screen.getByRole('textbox')).toBeDefined();
        expect(screen.getByRole('button')).toBeDefined();
    })


    test('Should call onQuery when form is submitted', async () => {
        const onQuery = vi.fn();
        render(<SearchBar onQuery={onQuery} />)
        const input = screen.getByRole('textbox');
        fireEvent.change(input, { target: { value: 'test' } });

        await waitFor(() => {
            expect(onQuery).toHaveBeenCalledTimes(1);
            expect(onQuery).toHaveBeenCalledWith('test');

        }
        );
    });

    test('Should call only once with the same value', async () => {
        const onQuery = vi.fn();
        render(<SearchBar onQuery={onQuery} />)
        const input = screen.getByRole('textbox');
        fireEvent.change(input, { target: { value: 't' } });

        await waitFor(() => {
            expect(onQuery).toHaveBeenCalledTimes(1);
            expect(onQuery).toHaveBeenCalledWith('t');
        })
    });

    test('Should call onQuery when button clicked with the input value', () => {
        const onQuery = vi.fn();
        render(<SearchBar onQuery={onQuery} />)
        const input = screen.getByRole('textbox');
        fireEvent.change(input, { target: { value: 't' } });

        const button = screen.getByRole('button');
        fireEvent.click(button);
        expect(onQuery).toHaveBeenCalledTimes(1);
        expect(onQuery).toHaveBeenCalledWith('t');

    });

    test('The input has the correct placeHolder', () => {
        render(<SearchBar onQuery={() => { }} placeholder="Buscar gif" />)
        expect(screen.getByPlaceholderText('Buscar gif')).toBeDefined();

    });


})