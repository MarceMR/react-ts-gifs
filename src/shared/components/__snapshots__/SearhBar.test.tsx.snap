// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en SearhBar > Should render component properly 1`] = `
<div>
  <div
    class="search-container"
  >
    <input
      placeholder="Buscar"
      type="text"
      value=""
    />
    <button>
      Buscar
    </button>
  </div>
</div>
`;
